engine/database/user_utxos.go

```diff
@@ -0,0 +1,15 @@
+package database
+
+import "time"
+
+// UserUtxos is a table holding user's Unspent Transaction Outputs (UTXOs).
+type UserUtxos struct {
+	XPubID                       string `gorm:"primaryKey;column:xpub_id;uniqueIndex:idx_window,sort:asc,priority:1"`
+	TxID                         string `gorm:"primaryKey;uniqueIndex:idx_window,sort:asc,priority:4"`
+	Vout                         uint32 `gorm:"primaryKey;uniqueIndex:idx_window,sort:asc,priority:5"`
+	Satoshis                     uint64
```

Comment:
Why don't we want to store `Satoshis` and `UnlockingScriptEstimatedSize` in the table? maybe move them as last variables if they are not "gorm related"